winch/codegen/src/codegen/mod.rs

```diff
@@ -1022,6 +1026,93 @@ where
         );
     }
 
+    /// Emit series of instructions that check the current epoch against its
+    /// deadline.
+    pub fn emit_epoch_check(&mut self) {
+        let cont = self.masm.get_label();
+        let new_epoch = self.env.builtins.new_epoch::<M::ABI, M::Ptr>();
+
+        // Checks for runtime limits (e.g., fuel, epoch) are special since they
+        // require inserting abritrary function calls and control flow.
+        // Special care must be taken to ensure that all invariants are met. In
+        // this case, since `new_epoch` takes an argument and returns a value,
+        // we must ensure that any registers used to hold the current epoch
+        // value and deadline are not going to be needed later on by the
+        // function call.
+        let (epoch_deadline_var, epoch_counter_var) = self.context.without::<(Reg, Reg), M, _>(
```

Comment:
Is the `_var` suffix fairly established in Winch? If so, ignore the following.

If not, then I find it a little surprising. I thought maybe there was a `Var` type that abstracts over registers and stack slots or something. Or introduces a virtual register concept or something like that. I think a `_reg` suffix would have avoided me going down that line of thought and inquiry, which would make it slightly easier to follow what's going on, as a code reader.